sources/api/early-boot-config/src/provider/local_file.rs

```diff
@@ -0,0 +1,68 @@
+//! The local_file module implements the `PlatformDataProvider` trait for gathering userdata from
+//! local file
+
+use super::{PlatformDataProvider, SettingsJson};
+use snafu::{OptionExt, ResultExt};
+use std::fs;
+
+pub(crate) struct LocalFileDataProvider;
+
+impl LocalFileDataProvider {
+    pub(crate) const USER_DATA_FILE: &'static str = "/etc/early-boot-config/user-data";
+}
+
+impl PlatformDataProvider for LocalFileDataProvider {
+    fn platform_data(&self) -> super::Result<Vec<SettingsJson>> {
+        let mut output = Vec::new();
+        info!("'{}' exists, using it", Self::USER_DATA_FILE);
+
+        let user_data_str =
+            fs::read_to_string(Self::USER_DATA_FILE).context(error::InputFileRead {
+                path: Self::USER_DATA_FILE,
+            })?;
+
+        if user_data_str.is_empty() {
+            return Ok(output);
+        }
+
+        // Remove outer "settings" layer before sending to API
```

Comment:
Repeated code, maybe keep an eye out for a third appearance and pull it if that happens.